src/model/channel/message.rs

```diff
@@ -140,14 +140,18 @@ impl Message {
         #[cfg(feature = "cache")]
         {
             if let Some(cache) = cache_http.cache() {
-                let permissions = Permissions::MANAGE_MESSAGES;
-                let is_author = self.author.id == cache.current_user().await.id;
-
-                utils::user_has_perms_cache(cache, self.channel_id, self.guild_id, permissions)
-                    .await?;
-
-                if !is_author {
-                    return Err(Error::Model(ModelError::NotAuthor));
+                if self.author.id != cache.current_user().await.id {
```

Comment:
This could use [`Cache::current_user_id`](https://serenity-rs.github.io/serenity/current/serenity/cache/struct.Cache.html#method.current_user_id) to prevent cloning the whole `CurrentUser` struct.
```suggestion
                if self.author.id != cache.current_user_id().await {
```